xonsh/built_ins.py

```diff
@@ -568,17 +568,24 @@ def run_subproc(cmds, captured=True):
         return
     if prev_is_proxy:
         prev_proc.wait()
-    wait_for_active_job()
+    job = wait_for_active_job()
+    output = None
     if write_target is None:
         # get output
-        output = ''
+        out = ''
         if prev_proc.stdout not in (None, sys.stdout):
-            output = prev_proc.stdout.read()
+            out = prev_proc.stdout.read()
         if captured:
-            return output
+            output = out
     elif last_stdout not in (PIPE, None, sys.stdout):
         last_stdout.close()
 
+    if job and job['obj'].returncode != 0 and builtins.__xonsh_ctx__['__xonsh_subproc_check__']:
+        cmd = [' '.join(i) if isinstance(i, list) else i for i in job['cmds']]
+        cmd = ' '.join(cmd)
+        raise CalledProcessError(job['obj'].returncode, cmd, output)
+    return output
```

Comment:
I think we only want to return output if we are captured.  Otherwise, we should be returning None.